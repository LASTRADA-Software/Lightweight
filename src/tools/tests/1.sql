CREATE TABLE "Person" (
    "id" INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
    "name" VARCHAR(25) NOT NULL,
    "is_active" BOOLEAN NOT NULL,
    "age" INTEGER
);
