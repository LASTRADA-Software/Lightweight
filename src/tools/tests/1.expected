// SPDX-License-Identifier: Apache-2.0
#pragma once
#include <Lightweight/DataMapper/DataMapper.hpp>
#include <Lightweight/SqlConnection.hpp>
#include <Lightweight/SqlDataBinder.hpp>
#include <Lightweight/SqlQuery.hpp>
#include <Lightweight/SqlQueryFormatter.hpp>
#include <Lightweight/SqlScopedTraceLogger.hpp>
#include <Lightweight/SqlStatement.hpp>
#include <Lightweight/SqlTransaction.hpp>

struct Person;

struct Person final
{
    Field<int, PrimaryKey::ServerSideAutoIncrement> id;
    Field<std::string> name;
    Field<bool> is_active;
    Field<std::optional<int>> age;
};
